### PR TITLE
Minor mbtiles summary improvements

### LINKED ISSUE #1002: Minor mbtiles summary improvements
The new `mbtiles summary` tool could have a bit of cleanup:

* [ ] Move all summary code from `mbtiles/src/mbtiles.rs` to `mbtiles/src/summary.rs`
* [ ] Use one of the pretty table printers to display the output. See [available crates](https://crates.io/search?q=category%3Acommand-line-interface%20format%20table&sort=recent-downloads).

The table printer is a bit of a tricky one - at first i was thinking of using [prettytable-rs](https://crates.io/crates/prettytable-rs) or [cli-table](https://crates.io/crates/cli-table) or something similar, but then I saw [termimad](https://crates.io/crates/termimad) -- which is a markdown renderer.  In theory, that could prove a much better approach (data will be represented as markdown internally, and can be rendered as either text or pretty-printed).  So some research is needed.

### PR DESCRIPTION
**Pull request by sharkAndshark**
Try to fix #1002 

* [x] Move all summary code from `mbtiles/src/mbtiles.rs` to `mbtiles/src/summary.rs`
* [x] Move Metadata and Validation function to separate files
* [x] Remove `filename` String from the summary stats - not much point because it is accessible from the mbtiles struct itself

### CODE REVIEW

File: mbtiles/src/util.rs
Comment (sharkAndshark) [inline mbtiles/src/util.rs:1]: Maybe move all things in `util.rs` to `summary.rs` better? @nyurik

**Comment by nyurik** [comment]
you should move the `summary()` function to the summary.rs too.  More than one file can add functions to the same struct:

```rust
# add this to summary.rs
use crate::mbtiles::Mbtiles;
impl Mbtiles {
    pub fn summary(&self, ...) {}
}
```

File: mbtiles/src/util.rs
Comment (nyurik) [reply mbtiles/src/util.rs:1]: I don't think you need util.rs just yet. All these functions are for just one usage -- summary.  Utils will be needed once more than one file will share common utilities.

Comment (sharkAndshark) [reply mbtiles/src/util.rs:1]: This feature is so great! I will give it a try!

**Comment by sharkAndshark** [comment]
> you should move the `summary()` function to the summary.rs too.  More than one file can add functions to the same struct:
> 
> 
> 
> ```rust
> 
> # add this to summary.rs
> 
> use crate::mbtiles::Mbtiles;
> 
> impl Mbtiles {
> 
>     pub fn summary(&self, ...) {}
> 
> }
> 
> ```

Great feature!

File: mbtiles/src/summary.rs [hunk 1]
```diff
@@ -0,0 +1,370 @@
...
+    pub avg_tile_size: f64,
+    pub bbox: Option<Bounds>,
+    pub min_zoom: Option<u8>,
+    pub max_zoom: Option<u8>,
+    pub zoom_info: Vec<ZoomInfo>,
+}
+
+impl Display for Summary {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        writeln!(f, "File: {}", self.file_path)?;
```
Comment (sharkAndshark) [inline mbtiles/src/summary.rs]: To access `file_path` in `summary.rs`, It's necessary to set `file_path` public.

Comment (nyurik) [reply mbtiles/src/summary.rs]: Set it to `pub(crate)`

Comment (sharkAndshark) [reply mbtiles/src/summary.rs]: Seems it's not bad to make it public.   
Another option it's to create a directory named`mbtiles` with two files:
* `mbtiles/mod.rs`
* `mbtiles/summary.rs`

@nyurik   Any help? Is it to make `file_path` public ok? I prefer  this approch as it's simple and it make sense to expose `file_path` ?

Comment (nyurik) [reply mbtiles/src/summary.rs]: Sure I'll look at the whole pr soon. Need to get some sleep 😀

Comment (sharkAndshark) [reply mbtiles/src/summary.rs]: Ok I need to handle some daily job now. See you soon. ：)

**Comment by nyurik** [comment]
@sharkAndshark I made a few changes to your branch - moving metadata and validation code also to separate files, and doing a few minor summary fixes. Let me know if you see any issues with those changes.

**Comment by nyurik** [comment]
I'll merge this as is for now, and the table view could go as a separate PR.  Thanks for all the hard work!!!

* [ ] Use one of the pretty table printers to display the output. See [available crates](https://crates.io/search?q=category%3Acommand-line-interface%20format%20table&sort=recent-downloads).

**Review by nyurik — APPROVED** [review]

**Pull request merged**